### PR TITLE
Compile macOS Intel Version on Apple Silicon

### PR DESCRIPTION
**Pull request by halprin**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

Eventually the GitHub runner `macos-13` will be deprecated and removed by GitHub.  This work is to get the macOS Intel version to compile with the Apple Silicon `macos-14` runner.  This work is also the first step towards making a universal (fat) binary that contains both Apple Silicon and Intel architectures in one binary.

### Related Issues

_None_.

### Screenshots (if applicable)

_None_.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### Additional Comments

_None_.

### CODE REVIEW

**Comment by halprin** [comment]
It works!  The Intel version continues to contain Intel binaries, the Apple Silicon version continues to contain Apple Silicon binaries, and both run on my Apple Silicon based Mac.   This is now ready for review.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -60,12 +60,27 @@ jobs:
...
+        if: ${{ matrix.os.name == 'macOS-Intel' }}
+        shell: arch -x86_64 /bin/bash -e {0}
         run: |
-          # Install packages from Homebrew
+          eval "$(/usr/local/bin/brew shellenv)"  # makes the Intel version of Brew and its installs are enshrined as the default 
+          echo "PATH=$PATH" >> $GITHUB_ENV  # modify $PATH so the above step persists into the future steps
           brew bundle install
 
+      - name: Install macOS ARM dependencies
+        if: ${{ matrix.os.name == 'macOS-ARM' }}
```
Comment (Lgt2x) [inline .github/workflows/build.yml:81]: we need to run `brew bundle install` for both builds, I'd suggest removing this condition as well as l.78 that does the same thing for intel

Comment (halprin) [reply .github/workflows/build.yml]: @Lgt2x, fixed and ready for re-review!

I thought that I had to also run `brew bundle install` with the specific x86_64 shell when compiling for Intel, but apparently all that is necessary is to install the Intel version of Brew and use it.

**Review by Lgt2x — APPROVED** [review]
Good job! I can't test it but I'll trust you that the generated binaries work properly :)

**Comment by Lgt2x** [comment]
woops I forgot to squash... too bad

**Pull request merged**